index.js

```diff
@@ -16,10 +17,13 @@ function AssetFingerprint(initializerDirectory, needsFingerprint = true) {
 AssetFingerprint.prototype.apply = function(compiler) {
   compiler.plugin('done', function(stats) {
     if (this.needsFingerprint) {
-      let output = `ASSET_FINGERPRINT = '${stats.hash}'`;
-      let initializerPath = `${this.initializerDirectory}/asset_fingerprint.rb`;
-
-      fs.writeFileSync(initializerPath, output);
+        let defaultRun = this.fingerprintName === 'ASSET_FINGERPRINT';
+        let initializerPath = `${this.initializerDirectory}/asset_fingerprint.rb`;
+        const newline = defaultRun ? '' : '\r\n';
+        const fsMethod = defaultRun ? 'writeFileSync' : 'appendFileSync';
```

Comment:
When you run this script more than once, it's going to append the custom constant multiple times. Is there a way we might be able to prevent that?

Reply:
I could read the file prior to writing the value back out.  If I find a string that matches the value for the name I could throw an exception.

Reply:
Is that what we'd want to do though? Or just no-op? The way it is (at least how it worked for me locally, which was what I was curious about when we talked last week) is that every subsequent build will just append a duplicate of the constant, or if files have changed, a new value for the same constant, similar to this:

```ruby
ASSET_FINGERPRINT='lkasd7fdjassdk23'
CUSTOM_ASSET_FINGERPRINT='lkasd7fdjassdk23'
CUSTOM_ASSET_FINGERPRINT='hhgdks553l2343n'
CUSTOM_ASSET_FINGERPRINT='hhgdks553l2343n'
```

Reply:
Ideally, the user should not call the plugin twice in the same build.  I would not want multiple instances of the same constant with the same or different value.